iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -120,14 +123,14 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     ///         value a new value is constructed with the copy constructor.
     /// @param[in] rhs source of the copy
     /// @return reference to the current optional
-    optional& operator=(const optional& rhs) noexcept;
+    optional& operator=(const optional& rhs) & noexcept;
```

Comment:
Please remove this! I know this is an autosar rule but this would actually make our C++ code non compliant to the c++ standard.

Take a look at: https://isocpp.org/files/papers/N4860.pdf Section 6.2 part 3.

The move and copy constructors are defined like:
```cxx
class C {
  C(const C& rhs); // copy ctor
  C(C&& rhs); // move ctor
  C& operator=(const C& rhs); // copy assignment
  C& operator=(C&& rhs); // move assignment
}
```

There is also no possible bug which can occur when the lvalue marker is missing.

Reply:
@elfenpiff Sure. Thanks